hack/demo-env/setup-vcluster-env.sh

```diff
@@ -20,29 +30,176 @@ namespace() {
+
+	# Update the Argo CD repo-server/redis addresses that agent-managed Argo CD instance connects to
+	sed -i.bak "s/repo-server-address/$REPO_SERVER_ADDR/g" "$TMP_DIR/agent-managed/argocd-cmd-params-cm.yaml"
+	sed -i.bak "s/redis-server-address/$REDIS_ADDR/g" "$TMP_DIR/agent-managed/argocd-cmd-params-cm.yaml"
+
 	echo "-> Creating Argo CD instances in vclusters"
-	for c in $VCLUSTERS; do
+	for c in $VCLUSTERS_AGENTS; do
 		cluster=$(cluster $c)
 		namespace=$(namespace $c)
 		echo "  --> Creating instance in vcluster $cluster"
 		kubectl --context vcluster-$cluster create ns $namespace || true
-		kubectl --context vcluster-$cluster apply -n $namespace -k ${SCRIPTPATH}/${cluster}
+
+		# Run 'kubectl apply' twice, to avoid error that occurs during the first invocation (see above for error)
+		kubectl --context vcluster-$cluster apply -n $namespace -k ${TMP_DIR}/${cluster} || true
+		kubectl --context vcluster-$cluster apply -n $namespace -k ${TMP_DIR}/${cluster}
 	done
+
 	kubectl --context vcluster-control-plane create ns agent-autonomous || true
 	kubectl --context vcluster-control-plane create ns agent-managed || true
 	kubectl --context vcluster-agent-managed create ns agent-managed || true
+
+	echo "-> Waiting for all the Argo CD/vCluster pods to be running on vclusters"
+	wait_for_pods vcluster-control-plane 5
+	wait_for_pods vcluster-agent-autonomous 4
+	wait_for_pods vcluster-agent-managed 4
+
 }
 
+check_for_openshift
+
+
 case "$action" in
 create)
+
+
+	kubectl create ns vcluster-agent-managed --context=${initial_context} || true
+	kubectl create ns vcluster-control-plane --context=${initial_context} || true
+	kubectl create ns vcluster-agent-autonomous --context=${initial_context} || true
+
+	EXTRA_VCLUSTER_PARAMS=""
+
+	if [[ "$OPENSHIFT" != "" ]]; then
+
+		# Ensure that the namespaces we are using for our vclusters use our custom SCC (see SCC yaml for details)
+		kubectl apply -f ${SCRIPTPATH}/resources/scc-anyuid-seccomp-netbind.yaml
+
+		oc adm policy add-scc-to-group anyuid-seccomp-netbind system:serviceaccounts:vcluster-agent-managed --context=${initial_context}
+
+		oc adm policy add-scc-to-group anyuid-seccomp-netbind system:serviceaccounts:vcluster-control-plane --context=${initial_context}
+
+		oc adm policy add-scc-to-group anyuid-seccomp-netbind system:serviceaccounts:vcluster-agent-autonomous --context=${initial_context}
+
+		EXTRA_VCLUSTER_PARAMS="-f ${SCRIPTPATH}/resources/vcluster.yaml"
+	fi
+
 	echo "-> Creating required vclusters"
 	for c in $VCLUSTERS; do
 		cluster=$(cluster $c)
-		echo "  --> Creating vcluster $cluster"
-		vcluster create --context=default -n vcluster-${cluster} --expose --kube-config-context-name vcluster-${cluster} vcluster-${cluster}
+ 		echo "  --> Creating vcluster $cluster"
+		vcluster create --context=${initial_context} ${EXTRA_VCLUSTER_PARAMS} --switch-context=false -n vcluster-${cluster} --expose --kube-config-context-name vcluster-${cluster} vcluster-${cluster}
```

Comment:
My vcluster version (0.19.6) doesn't support the `--switch-context` flag. Latest is 0.19.7, but changelog doesn't mention any added switch between .6 and .7.

```
-> Creating required vclusters
  --> Creating vcluster control-plane
15:01:00 fatal unknown flag: --switch-context
ERROR: Error occured, terminating.
```

Which vcluster version are you using?

Reply:
The vCluster docs imply one should use the beta versions by default:
https://www.vcluster.com/docs/vcluster/#deploy-vcluster

Thus I'm on:
```
[jgw@localhost-lan argocd-agent]$ vcluster -v
vcluster version 0.20.0-beta.12
```

Reply:
Cool, thanks. I think I will upgrade then, I thought that switch might have been from an older version. But if it's actually in the next stable version, I believe we can just leave it there.